lib/cfa/placer.rb

```diff
@@ -86,13 +94,16 @@ def initialize(matcher)
 
     # (see Placer#new_element)
     def new_element(tree)
-      index = tree.data.index(&@matcher)
-      res = {}
+      index = tree.data(filtered: false).index(&@matcher)
+      res = create_element
 
       if index
-        tree.data[index] = res
+        # remove old one and add new one, as it can have different key
+        # which cause problem to simple modify
```

Comment:
Did we have this problem even before, or only as a result of this more careful modification?
I still don't quite understand. Can we add a rspec test for this?